ref-exchange/src/account_deposit.rs

```diff
@@ -38,38 +109,50 @@ impl AccountDeposit {
     /// Subtract from `token` balance.
     /// Panics if `amount` is bigger than the current balance.
     pub(crate) fn sub(&mut self, token: &AccountId, amount: Balance) {
+        if *token == "" {
+            // We use empty string to represent NEAR
+            self.near_amount -= amount;
+            self.assert_storage_usage();
+            return;
+        }
         let value = *self.tokens.get(token).expect(ERR21_TOKEN_NOT_REG);
         assert!(value >= amount, "{}", ERR22_NOT_ENOUGH_TOKENS);
         self.tokens.insert(token.clone(), value - amount);
     }
 
-    /// Returns amount of $NEAR necessary to cover storage used by this data structure.
+    /// Returns amount of $NEAR necessary to cover storage used by account referenced to this structure.
     pub fn storage_usage(&self) -> Balance {
-        (MIN_ACCOUNT_DEPOSIT_LENGTH + self.tokens.len() as u128 * (MAX_ACCOUNT_LENGTH + 16))
-            * env::storage_byte_cost()
+        let s = self.storage_used
+            + INIT_ACCOUNT_STORAGE  // empty account storage
+            + (ACC_ID_STORAGE + U64_STORAGE) * self.tokens.len() as u64; // self.tokens storage
```

Comment:
why not just count this inside storage_used? so `storage_usage * cost` is just self.storage_used?
also it seems like you are missing cost of storage in general here.

Reply:
Because, in `self.storage_used` we only account storage used outside of this structure. This is to avoid "double" storing the account. See here: https://github.com/ref-finance/ref-contracts/pull/7#discussion_r603322958